### PR TITLE
feat: t16 - buildpacks - slow php reporting page

### PR DESCRIPTION
**Pull request by rtest12**
## Task 16 - Compile PHP app using paketo builder and paketo php buildpack

You should have finalized your slow php reporting page test task defined in `saritasa-devops-camp/chapter3 - tech-stack components/l2/l2 - php.md`

This time you will have to compile OCI image using `pack` cli and run container that will implement the solution of that task. https://paketo.io/docs/howto/php/

You will need to also use nginx buildpack https://paketo.io/docs/howto/web-servers/ https://paketo.io/docs/reference/nginx-reference/

Use paketobuildpacks/builder:full full builder for simplicity.

Your PR Name: buildpacks - slow php reporting page

- Inside your PR - put all files as required by the buildpack, including your PHP files.
- In summary display a sequence of all CLI calls and outputs.
- Prove with `time curl <URL>` that your functionality works as expected.

###  build
```
pack build phpslowbp --builder paketobuildpacks/builder:full --env BP_PHP_SERVER=nginx --buildpack gcr.io/paketo-buildpacks/php-nginx --buildpack paketo-buildpacks/php --buildpack paketo-buildpacks/procfile --default-process app
```
### my output:

[t16.log](https://github.com/rtest12/devops-camp-2023/files/11359426/t16.log)

### CODE REVIEW

File: chapter2/tests/t16/project.toml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+[project]
+id = "io.buildpacks.php-slow-script"
+name = "buildpacks - slow php reporting page"
+version = "1.0.0"
+BP_PHP_VERSION="8.1.7"
```
Comment (dmitry-mightydevops) [inline chapter2/tests/t16/project.toml]: incorrect, you should define it as ENV var
[BP_PHP_VERSION](https://paketo.io/docs/howto/php/)

https://github.com/buildpacks/spec/blob/main/extensions/project-descriptor.md#iobuildpacksbuildenv-optional

File: chapter2/tests/t16/project.toml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+[build]
+exclude = [
+    "README.md",
+    "*.toml"
+]
+
+# include = []
```
Comment (dmitry-mightydevops) [inline chapter2/tests/t16/project.toml]: remove if not needed

File: chapter2/tests/t16/project.toml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+[[ build.env ]]
+name='TEAM'
+value='DEVOPS'
+
+
+[[processes]]
+type = "app"
+command = "/workspace/run.sh"
```
Comment (dmitry-mightydevops) [inline chapter2/tests/t16/project.toml]: no need - why? You're using procfile.

File: chapter2/tests/t16/run.sh [hunk 4]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+# Task 11 - slow php reporting page
+set -eo pipefail
+
+# Start php-fpm in foreground mode
+php-fpm -F --fpm-config $(pwd)/.php.fpm.d/php-fpm.conf &
```
Comment (dmitry-mightydevops) [inline chapter2/tests/t16/run.sh]: I don't see .php.fpm.d folder. 
your php-fpm.conf should be placed in a proper place so the php-fpm buildpack kicks in.

File: chapter2/tests/t16/nginx-fpm.conf
Comment (dmitry-mightydevops) [inline chapter2/tests/t16/nginx-fpm.conf:1]: see how nginx paketo buildpack expects the configuration.

every buildpack is opionated - they look for specific files in specific folders (or the root) to make a decision (that's how their `detect` func is working)

see samples to understand more details
https://github.com/paketo-buildpacks/samples/tree/main/web-servers/nginx-sample

File: chapter2/tests/t16/run.sh (hunk 4 above)
Comment (dmitry-mightydevops) [reply chapter2/tests/t16/run.sh]: See more:
https://github.com/paketo-buildpacks/samples/tree/main/php

Comment (dmitry-mightydevops) [reply chapter2/tests/t16/run.sh]: your root level php-fpm.conf should include *.conf files in the .php.fpm.d folder that define slow.conf and fast.conf pools

**Review by dmitry-mightydevops — CHANGES_REQUESTED** [review]
your configuration is not correct - buildpack expects config files in specific folders with specific names. I don't see them.

for example your pack build cli is incorrect.
you only need to define the builder. 

everything else buildpack would figure out itself if you used proper naming/folder structure.

your php scripts should be in the root, not in the htdocs/**/** as that way buildpack can identify it is dealing with php code and php buildpack is needed.

File: chapter2/tests/t16/project.toml (hunk 1 above)
Comment (rtest12) [reply chapter2/tests/t16/project.toml]: > incorrect, you should define it as ENV var [BP_PHP_VERSION](https://paketo.io/docs/howto/php/)

@dmitry-mightydevops 
Done

File: chapter2/tests/t16/project.toml (hunk 2 above)
Comment (rtest12) [reply chapter2/tests/t16/project.toml]: > remove if not needed

@dmitry-mightydevops 
Done

File: chapter2/tests/t16/project.toml (hunk 3 above)
Comment (rtest12) [reply chapter2/tests/t16/project.toml]: > no need - why? You're using procfile.

@dmitry-mightydevops 
Done

File: chapter2/tests/t16/run.sh (hunk 4 above)
Comment (rtest12) [reply chapter2/tests/t16/run.sh]: > I don't see .php.fpm.d folder. your php-fpm.conf should be placed in a proper place so the php-fpm buildpack kicks in.

@dmitry-mightydevops 
Done

File: chapter2/tests/t16/nginx-fpm.conf
Comment (rtest12) [reply chapter2/tests/t16/nginx-fpm.conf:1]: > see how nginx paketo buildpack expects the configuration.
> 
> every buildpack is opionated - they look for specific files in specific folders (or the root) to make a decision (that's how their `detect` func is working)
> 
> see samples to understand more details https://github.com/paketo-buildpacks/samples/tree/main/web-servers/nginx-sample

@dmitry-mightydevops 
Done

**Comment by dmitry-mightydevops** [comment]
@rtest12 I will review this one tomorrow.
